### DISCUSSION TITLE
betrayal

### DISCUSSION THREAD

**Discussion by shprotru** [ihhub/fheroes2#5801] [General]
Good day, dear heroes2 fans! The map `Ghost Planet` is incorrectly processed by the original game: you can see that you are in alliance with Green, Red and Yellow players(by description), but you can attack them and you don't see their area of vision. In fheroes2 map loads correctly, but this map is special: the enemy near your start location own units `ghosts`, which make their amout increased after every attack on a weak units. And your yellow ally there is own 100 peasants, and in areas of reach of enemy with `ghosts`. If you spent your turn, the enemy can attack your ally and get too much ghosts, which makes this map unplayable.
I made PR #5790, which allows human players to attack allies(if it controlled by AI or a human). Before attack the game will ask you to confirm your intentions in attack ally. After attack that ally will be your enemy. You can already check yourself how it works there, in that PR.

Main quiestion of this discussion is: need this feature to AI or ther is no need in that?

My own answer is `solid no`, and that's why:
it is very difficult to assess there: if there is an enemy hero with a pack of `ghosts`, and the hero of an allied player has a large number of weak units and is in the reach of both allied and enemy heroes, then the AI must destroy him so that he does not get `ghosts`. And we also need to check that he has a stronger army than his ally, but at the same time he did not have many weak units.
See, too much conditions to check? This will make our AI slower.

But our map editor should have possibility to make AI betrayal different way: it should contain triggers for:
- on visiting some area
- attack some stack of neutrals
- object capturing
- after some turns passed

**Comment by ihhub** [comment]
Hi @shprotru , I believe that we must address the root of this issue which is described in #5792.
